lib/settings/settings_screen.dart

```diff
@@ -39,37 +47,195 @@ class SettingsScreen extends StatefulWidget {
+              },
+              child: const Text('Later', style: TextStyle(fontSize: 20)),
+            ),
+            const SizedBox(width: 20),
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop(true);
+              },
+              child: const Text('Now', style: TextStyle(fontSize: 20)),
+            ),
+          ],
+        );
+      },
+    );
+
+    if (shouldRestart) {
+      logger.info('Restarting Orion');
+      config.setFlag('needsRestart', false, category: 'internal');
+      setState(() {
+        needsRestart = false;
+      });
+      restartOrion();
+    }
+  }
+
+  Future<void> launchDisconnectDialog() async {
+    bool shouldDisconnect = await showDialog(
+      barrierDismissible: false,
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Disconnect from WiFi'),
+          content: const Text(
+              'Do you want to disconnect from the current WiFi network?\nThis may cause any ongoing print jobs to fail.'),
+          actions: [
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop(false);
+              },
+              child:
+                  const Text('Stay Connected', style: TextStyle(fontSize: 20)),
+            ),
+            const SizedBox(width: 20),
+            TextButton(
+              onPressed: () {
+                Navigator.of(context).pop(true);
+              },
+              child: const Text('Disconnect', style: TextStyle(fontSize: 20)),
+            ),
+          ],
+        );
+      },
+    );
+
+    if (shouldDisconnect) {
+      await _wifiScreenKey.currentState?.disconnect();
+    }
+  }
+
+  void restartOrion() async {
+    try {
+      final String screenRotation =
+          config.getString('screenRotation', category: 'advanced');
+      final String baseUser = Platform.environment['BASE_USER'] ?? 'pi';
+
+      // Load the bash script from assets
+      final String bashScript =
+          await rootBundle.loadString('assets/scripts/set_orion_config.sh');
+
+      // Create a temporary directory
+      final Directory tempDir = await Directory.systemTemp.createTemp();
+      final String scriptPath = path.join(tempDir.path, 'set_orion_config.sh');
+
+      // Write the bash script to the temporary file
+      final File scriptFile = File(scriptPath);
+      await scriptFile.writeAsString(bashScript);
+
+      // Make the script executable
+      await Process.run('chmod', ['+x', scriptPath]);
+
+      // Execute the script with sudo
+      final result =
+          await Process.run('sudo', [scriptPath, screenRotation, baseUser]);
+
+      if (result.exitCode != 0) {
+        logger.severe('Failed to restart Orion: ${result.stderr}');
+      } else {
+        logger.info('Orion restarted successfully');
+      }
+
+      // Clean up the temporary file
+      await scriptFile.delete();
```

Comment:
Is there a reason we can't just execute the script from the assets dir?

Reply:
I haven't got that approach to work yet, but will try for the next release.